requirements/pyinstaller-build.txt

```diff
@@ -1,3 +1,5 @@
 # Executable binary builder requirements
-setuptools==70.3.0
+
+backports.tarfile==1.2.0; python_version=="3.8"
```

Comment:
Is this permanent? Can we also drop a comment here about why its required?

Reply:
Yeah, can leave a comment here.

Reply:
@hnnasit were you able to find why we need this?

Reply:
Left a comment about what I understand the problem is here. We could also run dev canaries against this PR to make sure nothing else breaks.